provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -890,6 +890,17 @@ func generateSchema() schema.PackageSpec {
 							"Valid values are `AL2`, `AL2023` and `Bottlerocket`.\n\n" +
 							"Defaults to `AL2`.",
 					},
+					"bottlerocketSettings": {
+						TypeSpec: schema.TypeSpec{
+							Type: "object",
+							AdditionalProperties: &schema.TypeSpec{
+								Ref: "pulumi.json#/Any",
+							},
+						},
+						Description: "The configuration settings for Bottlerocket OS.\n" +
+							"The settings will get merged with the base settings the provider uses to configure Bottlerocket.\n\n" +
+							"For an overview of the available settings, see https://bottlerocket.dev/en/os/1.20.x/api/settings/.",
```

Comment:
Typifying https://bottlerocket.dev/en/os/1.20.x/api/settings-index/ seems possible with some legwork. There seems to be an OpenAPI definition somewhere.

Reply:
I did look into this, but they do not have any types for the actual configuration options: https://github.com/bottlerocket-os/bottlerocket/blob/a3e8bda116dcf4b06fdf6e4cea42da1a005e926f/sources/api/openapi.yaml#L108-L109

There's also the problem of choosing the right version of the typed object for the right bottlerocket version. E.g. there might be a change in the bottlerocket config, we'd have to accept all different versions of bottlerocket config. This would essentially cause us to fall back to accepting anything and we need an escape hatch, otherwise users will always have to wait for us to release updates.

So IMO this wouldn't really bring any benefits, rather complications for users.
My reasoning would be different if bottlerocket offered rich types though.